src/bootstrap/src/core/config/config.rs

```diff
@@ -1325,7 +1325,11 @@ impl Config {
         // Give a hard error if `--config` or `RUST_BOOTSTRAP_CONFIG` are set to a missing path,
         // but not if `config.toml` hasn't been created.
         let mut toml = if !using_default_path || toml_path.exists() {
-            config.config = Some(toml_path.clone());
+            config.config = Some(if cfg!(not(feature = "bootstrap-self-test")) {
```

Comment:
Maybe we can just use https://doc.rust-lang.org/stable/std/path/fn.absolute.html instead? Unless you also wanted to check here that the path exists.

Reply:
`canonicalize` also helps to resolve symlinks.

Reply:
Good point, let's keep it then.